content/courses/program-optimization/rust-macros.md

```diff
@@ -573,7 +573,7 @@ macro.
 #[derive(Accounts)]
 #[instruction(input: String)]
 pub struct Initialize<'info> {
-    #[account(init, payer = payer, space = 8 + input.len())]
+    #[account(init, payer = payer, space = MyData::DISCRIMINATOR.len() + MyData::INIT_SPACE + input.len())]
```

Comment:
~Ah I'd always made a `usize` constant set to 8. I guess I can get it from the `struct` instead? Maybe it will work better when Anchor switches to variable length discriminators. What do you think here?~